### DISCUSSION TITLE
DirectionalLight direction

### DISCUSSION THREAD

**Discussion by strattonbrazil** [bevyengine/bevy#5186] [Q&A] [answered]
Can I get some advice on how the direction of the DirectionalLight is controlled using its entity's transform? 

Right now I'm just creating the light over an XY plane and some cubes, and starting it with the following transform:
```
        transform.translation = Vec3::new(0.0, 0.0, 0.0);
        transform.rotation = Quat::from_rotation_x(-std::f32::consts::FRAC_PI_4);
```

![image](https://user-images.githubusercontent.com/786471/177048349-897fc143-4e81-4a47-920f-b9a9b4ce48f4.png)

I'm assuming the direction is calculated from some point with the rotation applied as the translation doesn't seem to do anything. I've tried animating the rotation to understand what's happening, but wasn't understanding why it was behaving that way. I'm assuming it has something to do with a hard-coded up-vector somewhere. 
```
fn set_light(time: Res<Time>, mut query: Query<(&mut OrbitingLight, &mut Transform)>) {
    for (mut orbiting_light, mut transform) in query.iter_mut() {
        transform.translation = Vec3::new(0.0, 0.0, 0.0);
        transform.rotation = Quat::from_rotation_x(-std::f32::consts::FRAC_PI_4 + (0.5 * time.seconds_since_startup() as f32));
    }
```

![ezgif-4-525dfe00a3](https://user-images.githubusercontent.com/786471/177048966-d2d06a03-2d17-4c95-a382-c4f8f4c2b2cd.gif)

**Comment by superdump** [comment, marked as answer]
Directional light sources are modelled to be at infinity and have parallel rays. As such they do not have a position in practical terms and only the rotation matters. The direction of the light is defined by the forward direction and by default the -z axis is forwards (right-handed, y-up, z points backwards and -z is forwards). Rotations are then applied to a Vec3 of (0,0,-1) to calculate the transform’s forward direction. Shadows are more complicated and not so usable yet for directional lights but I am finishing off an automatic shadow implementation for directional lights right now.

**Reply by strattonbrazil** [reply]
Thanks, @superdump. Is the default documented somewhere? I saw some context in the [struct doc](https://docs.rs/bevy/latest/bevy/prelude/struct.DirectionalLight.html). Is it kosher to add a line to that doc to explain the default direction?

**Reply by superdump** [reply]
That’s definitely a good idea. I went to look there to see if it was documented and then went digging in the code to see what was done. A PR would be welcome, to clarify the light direction in the documentation there as being explicitly in the forward direction of the GlobalTransform.

**Reply by strattonbrazil** [reply]
https://github.com/bevyengine/bevy/pull/5188
